### ISSUE TITLE
npm run build fails on frontend setup

### ISSUE THREAD

**Issue by Ozzah** [benreid24/Stellaru#155]
I'm following the instructions in the Readme for the "Running From Source" -> "One Time Setup".

My environment is Node.js 18.19.0 and npm 9.2.0, both of which are current and stable as at today. I'm running Python 3.10.14. My `uname -a` is:

`Linux f1cd504fc97e 6.8.0-39-generic #39-Ubuntu SMP PREEMPT_DYNAMIC Fri Jul  5 21:49:14 UTC 2024 x86_64 GNU/Linux`

Here are the outputs of `npm install` and `npm run build` respectively:
[npm_install.log](https://github.com/user-attachments/files/16401906/npm_install.log)
[npm_run_build.log](https://github.com/user-attachments/files/16401908/npm_run_build.log)

Full context: I am trying to set up a Docker file to pull, build, and run Stellaru, which I plan to make available either as a PR here or as a separate repo. Here are the steps leading up to the error:

```Dockerfile
FROM python:3.10-slim

RUN apt-get update && apt-get install -y \
    build-essential git npm \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

RUN git clone https://github.com/benreid24/Stellaru.git /app

WORKDIR /app/frontend
RUN npm install
RUN npm run build
```

**Comment by benreid24** [comment]
I am taking a look. A lot of the dependencies (both backend and frontend) were created a couple of years ago and haven't been updated since. I will try to upgrade everything and make sure the project still builds and works as expected. A Docker file would be appreciated!

**Comment by benreid24** [comment]
Landed a bunch of changes which should unblock you. Everything works for me locally on Windows using the same node version. I also beefed up the PR checks to test all three environments so hopefully it should build for you now.

**Comment by Ozzah** [comment]
Hi. Thank you, I confirm the latest version proceeds beyond this step.

I'm now encountering an issue with the `xdg` package where `XDG_DATA_HOME` is not found in [linux_finder.py](https://github.com/benreid24/Stellaru/blob/master/backend/engine/finders/linux_finder.py#L2).

When I have time, I will investigate further whether this is to do with my individual installation, or if this is a deprecated component of `xdg`. If so, I can raise another issue.

**Comment by Ozzah** [comment]
Regarding the above, have you seen: https://github.com/srstevenson/xdg-base-dirs/commit/3019d7dfe750965d332dd32f2aff6c59bf02b1d0

It looks like there was a collision between two different packages called `xdg`, so about 18 months ago one was renamed to `xdg-base-dirs`.
